src/Bedrock.Framework.Experimental/Transports/WebSockets/WebSocketPayloadReader.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Bedrock.Framework.Protocols;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO.Pipelines;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Bedrock.Framework.Experimental.Transports.WebSockets
+{
+    /// <summary>
+    /// Reads a WebSocket payload from a sequence.
+    /// </summary>
+    public struct WebSocketPayloadReader : IMessageReader<ReadOnlySequence<byte>>
+    {
+        /// <summary>
+        /// The number of bytes remaining in the payload.
+        /// </summary>
+        private ulong _payloadBytesRemaining;
+
+        /// <summary>
+        /// Whether or not to use SIMD acceleration for payload unmasking.
+        /// </summary>
+        private bool _useSimd;
```

Comment:
Remove this -- same as above. Also reduces the size of the struct (by one byte :wink:).

Reply:
This one is a bit of a trade-off - the idea here is to store whether or not to use SIMD in order to avoid having to do that lookup on every call to TryParseMessage. I'm not sure which would be better, might be worth benchmarking when I get to that point.

Reply:
Let the benchmarks decide. We should add some of those too so that we have a baseline for improvements.

Reply:
> having to do that lookup on every call to TryParseMessage

You mean the test for `Vector.IsHardwareAccelerated`?
The JIT will treat this as constant, hence it removes dead branches.

If SIMD is used / implemented / whatever should be treated as implementation detail solely, so it shouldn't show up -- even in internal / private members.
What if in some hypothetical future SIMT will be supported? Or any other technology.

When you are concerned about the additional length check, then you exchange that one with the check for `if (_useSimd)` on this should be equivalent.

So I don't expect any perf-win, and I would go with the more straightforward and direct code.
(the one without this extra field, argument).
Just test for SIMD and length where it's needed.